### DISCUSSION TITLE
new Code of Conduct Working Group seeking volunteers

### DISCUSSION THREAD

**Discussion by xmunoz** [coin-or/COIN-OR-OptimizationSuite#16] [General]
A few weeks ago I opened an issue (https://github.com/coin-or/CoinUtils/issues/172) about adding a Code of Conduct (CoC) to a COIN-OR repository. Another COIN-OR contributor took this issue to the board, and there was consensus that the [Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) would be a good starting point, and should be adopted across all repositories in the organization. If you have any feedback or concerns about COIN-OR adopting the Contributor Covenant as the default CoC, please comment on that issue.

While we wait for more feedback about rolling out the [Contributor Covenant](https://www.contributor-covenant.org/version/2/0/code_of_conduct/), I'd like to get a sense of who might be interested in volunteering in a working group dedicated to the enforcement of a prospective CoC, as well as expanding or updating the content in the selected CoC. This is anticipated to be a low-commitment opportunity primarily focused around ensuring that COIN-OR remains a friendly and inclusive community. 

Volunteers will be added to a mailing list where community members can report CoC violations. They will then investigate those alleged violations and follow up with the reporter and reported person to communicate remedies.This is all entirely new for COIN-OR, but other open-source organizations have adopted similar practices and have published helpful resources for addressing CoC violations (e.g. handling conflicts of interests, reporting procedures, etc). I found the following guides to be particularly illuminating:
- https://www.mozilla.org/en-US/about/governance/policies/participation/reporting
- https://www.python.org/psf/conduct/enforcement/

This is all to say that this group won't be starting completely from scratch, and will be able to flexibly craft policies in response to issues as they arise. No prior experience with code of conducts or anything else in the above text is required; just a willingness to address community concerns, and the ability to respond within a reasonable time frame.

Thanks for reading and please post below if you'd like to volunteer or have any questions!

**Comment by svigerske** [comment]
I don't mind if the COIN-OR organization or the COIN-OR Optimization Suite (which are two very different things) wants to adopt this, but I don't want to add such a code-of-conduct file to the projects I maintain (one of them being in the COIN-OR Optimization Suite).
I find the pledge and standards in this covenant self-evident and am not aware of gross violations for projects I am involved with. There was also no need to admonish or ban people so far, nor did I ever notice someone saying that they wouldn't contribute because there is no code of conduct explicitly written down. As you say, COIN-OR is a friendly and inclusive community. I would find it a bit unsettling if this wouldn't be clear without a written statement.

If the COIN-OR organization wants to adopt this and puts this onto its website (www.coin-or.org) or the COIN-OR Optimization Suite wants to adopt it and puts this into its repo, I wouldn't mind linking to it in the README or documentation of projects I maintain, but I prefer not to add files that I find unnecessary.   
Of course, it is not my business if other projects wants to add this. That's why I approved your pull request in CoinUtils - and I appreciated that you checked with the contributors of CoinUtils.

**Comment by xmunoz** [comment]
> I wouldn't mind linking to it in the README or documentation of projects I maintain, but I prefer not to add files that I find unnecessary.

Thanks for the suggestion, I think this is a great idea! Since most of these README's are auto-generated by coinbrew, this would definitely be the most efficient way to roll it out across the organization.

> I find the pledge and standards in this covenant self-evident and am not aware of gross violations for projects I am involved with. 

I am not either. The impetus for taking this action now is:
1. There is no explicitly defined Code of Conduct for COIN-OR, and I think that something is better than nothing
2. COIN-OR has expressed interest in pursuing support and funding from organizations that require a code of conduct

This isn't meant to be punitive or reactive, but rather strategic in helping further the goals of the COIN-OR Foundation.

**Reply by svigerske** [reply]
> > I wouldn't mind linking to it in the README or documentation of projects I maintain, but I prefer not to add files that I find unnecessary.
> 
> Thanks for the suggestion, I think this is a great idea! Since most of these README's are auto-generated by coinbrew, this would definitely be the most efficient way to roll it out across the organization.

There are 67 projects listed at https://www.coin-or.org/projects/ at the moment. For a few prominent, the README is autogenerated, but I don't think that it qualifies for "most" :).

**Reply by xmunoz** [reply]
Ah ok, thanks for the correction :)

**Reply by tkralphs** [reply]
Just to add a bit more context, the board of the COIN-OR Foundation is currently in the process of adopting this code of conduct for the entire organization, which would mean individual projects would not need to manage this on their own. Indeed, as suggested, the current plan is to post a document centrally on the COIN-OR Web site. Individual projects would then just need to link to this centrally maintained document from a README. 

Personally, I think there are good reasons to adopt such a code, even if it seems self-evident and even if it appears to us that there has been no need for such a mechanism so far. As an organization with unmoderated forums open to the public, we are always susceptible to bad actors who do not abide the community standards we may feel we already implicitly adhere to. All in all, it seems prudent to be a little more explicit about what our community standards and expectations are before there is actually an issue. Addressing these kinds of issues in hindsight is never good. It does also serve to communicate our values more clearly to people who may be wondering if our community is one they want to contribute to. This might encourage people who wouldn't otherwise participate to do so.

**Comment by hhijazi** [comment]
Thanks for the initiative @xmunoz, I agree that something is better than nothing, I would be happy to help here. As @svigerske pointed out, we hopefully won't have a lot of work since COIN-OR is already quite friendly :) I also prefer adding a link in the Readme as opposed to a file.

**Reply by xmunoz** [reply]
Hi @hhijazi, thanks for volunteering! Could you send an email to cristina@coin-or.org with the email address you'd like to add to this new mailing list? We're aiming to get this rolled out in the next week or two.
